noisepagecontrol/exploratory_worker/views.py

```diff
@@ -1,5 +1,70 @@
-from django.http import HttpResponse
+import json
+import subprocess
+import pathlib
+import os
+from threading import Thread
+from django.http import HttpResponse, HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+
+DIR_PATH = pathlib.Path(__file__).resolve().parent
+START_EXPLORATORY_SCRIPT = os.path.join(DIR_PATH, 'scripts/start_exploratory_cluster.sh')
+STOP_EXPLORATORY_SCRIPT = os.path.join(DIR_PATH, 'scripts/stop_exploratory_cluster.sh')
 
 
 def index(request):
     return HttpResponse("Hello, world. This is the exploratory_worker")
+
+
+@csrf_exempt
+@require_http_methods(["POST"])
+def launch_exploratory_cluster(request):
+    snapshot = json.loads(request.body)['snapshot']
+    # asynchronously spins up a pg instance
+    thread = Thread(target=start_exploratory_cluster, args=(snapshot,))
+    thread.start()
+
+    return HttpResponse()
+
+
+def start_exploratory_cluster(snapshot):
```

Comment:
Let's refactor this out into a new service; Check how primary_worker is organised